### PR TITLE
Partially fixes and verifies network routes

### PR DESCRIPTION
**Pull request by bkase**
Specifically, list + status are behaving to spec.

Introduced a simple `test/` system for checking changes.

Tested by running the two new test bash scripts against the debug network.

#5307

### CODE REVIEW

File: src/app/rosetta/network.ml [hunk 1]
```diff
@@ -29,36 +30,77 @@ module Get_version = [%graphql {|
...
+      peers
+    }
+    initialPeers
+  }
+|}]
+
 let genesis_block_query =
   Caqti_request.find Caqti_type.unit Caqti_type.string
-    "SELECT state_hash FROM blocks WHERE height = 0"
+    "SELECT state_hash FROM blocks WHERE height = 1 LIMIT 1"
```
Comment (lk86) [inline src/app/rosetta/network.ml:46]: Is the state_hash on block 1 equal to state_hash of block 0? I don't understand why these queries are semantically equivalent

File: src/app/rosetta/network.ml [hunk 2]
```diff
@@ -29,36 +30,77 @@ module Get_version = [%graphql {|
...
         Deferred.return
           ( match res#bestChain with
-          | Some [||] ->
-              Error (Errors.create "No blocks in chain")
+          | None | Some [||] ->
+              Error
+                (Errors.create
+                   "Could not get chain information. This probably means you \
+                    are bootstrapping -- bootstrapping is the process of \
+                    synchronizing to peers that are way ahead of you on the \
```
Comment (lk86) [inline src/app/rosetta/network.ml:100]: Grammatically I would use "synchronizing /with/ peers" here

File: src/app/rosetta/network.ml (hunk 1 above)
Comment (bkase) [reply src/app/rosetta/network.ml:46]: height 1 is the genesis block -- the height=0 query that was here before was untested (and broken).

Comment (bkase) [reply src/app/rosetta/network.ml:46]: ugh I guess it has to be 0 to be consistent with other blockchains, I'll open a bug to fix this

**Pull request merged**